catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.java

```diff
@@ -45,15 +48,19 @@
 	private String titleActionBar;
 	private boolean returnByPressingBackButton;
 
+	public static final String RECOVERED_FROM_CRASH = "RECOVERED_FROM_CRASH";
+	private static SharedPreferences sharedPreferences;
+
 	@Override
 	protected void onCreate(Bundle savedInstanceState) {
 		super.onCreate(savedInstanceState);
 		titleActionBar = null;
 		returnToProjectsList = false;
 		returnByPressingBackButton = false;
 		Thread.setDefaultUncaughtExceptionHandler(new BaseExceptionHandler());
-		Utils.checkIfCrashRecoveryAndFinishActivity(this);
+		checkIfCrashRecoveryAndFinishActivity(this);
 		getActionBar().setDisplayHomeAsUpEnabled(true);
+		sharedPreferences = PreferenceManager.getDefaultSharedPreferences(this);
```

Comment:
https://www.jetbrains.com/help/idea/inline.html

However, the inline refactoring of Android Studio doesn't work here.
Here by inline I mean replace `sharedPreferences` with `PreferenceManager.getDefaultSharedPreferences(this)` everywhere.

I think we don't need a member nor class member sharedPreferences. The only method sharedPreferences is used is in checkIfCrashRecoveryAndFinishActivity.

Since sharedPreferences != null can be removed as well, you can either remove isRecoveredFromCrash or pass in the sharedPreferences as parameter.